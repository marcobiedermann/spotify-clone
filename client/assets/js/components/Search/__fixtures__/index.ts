const search = {
  onSubmit: (): void => console.log('Searching â€¦'),
};

export default search;
