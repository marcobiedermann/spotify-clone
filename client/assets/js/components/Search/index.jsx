import PropTypes from 'prop-types';
import React from 'react';
import styles from './style.css';

const Search = ({ onSubmit }) => (
  <form
    className={styles.search}
    onSubmit={onSubmit}
  >
    <input
      className={styles.search__input}
      type="search"
    />
  </form>
);

Search.propTypes = {
  onSubmit: PropTypes.func,
};

Search.defaultProps = {
  onSubmit: () => {},
};

export default Search;
