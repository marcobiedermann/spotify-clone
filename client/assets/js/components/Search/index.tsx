import classNames from 'classnames';
import React, { FC } from 'react';
import styles from './style.css';

export interface SearchProps {
  className?: string;
  onSubmit: () => void;
}

const Search: FC<SearchProps> = (props) => {
  const { className, onSubmit } = props;

  return (
    <form className={classNames(className, styles.search)} onSubmit={onSubmit}>
      <input className={styles.search__input} type="search" />
    </form>
  );
};

export default Search;
