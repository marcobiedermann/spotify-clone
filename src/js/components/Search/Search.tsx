import clsx from 'clsx';
import styles from './Search.module.css';

export interface SearchProps {
  className?: string;
  onSubmit: () => void;
}

function Search(props: SearchProps): JSX.Element {
  const { className, onSubmit } = props;

  return (
    <form className={clsx(className, styles.search)} onSubmit={onSubmit}>
      <input className={styles.search__input} type="search" />
    </form>
  );
}

export default Search;
