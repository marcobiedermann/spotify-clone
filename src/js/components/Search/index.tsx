import classNames from 'classnames';
import React from 'react';
import styles from './style.module.css';

export interface SearchProps {
  className?: string;
  onSubmit: () => void;
}

function Search(props: SearchProps): JSX.Element {
  const { className, onSubmit } = props;

  return (
    <form className={classNames(className, styles.search)} onSubmit={onSubmit}>
      <input className={styles.search__input} type="search" />
    </form>
  );
}

export default Search;
