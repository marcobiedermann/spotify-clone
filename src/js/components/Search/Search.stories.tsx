import type { Meta, StoryObj } from '@storybook/react';
import Search from '.';

const meta: Meta<typeof Search> = {
  component: Search,
  args: {
    onSubmit: () => console.log('Searching â€¦'),
  },
};

type Story = StoryObj<typeof Search>;

const Default: Story = {};

export { Default };
export default meta;
