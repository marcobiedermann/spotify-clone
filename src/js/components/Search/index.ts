import Search from './Search';

export default Search;
export * from './Search';
