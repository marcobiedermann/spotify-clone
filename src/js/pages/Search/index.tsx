import { Helmet } from 'react-helmet';
import { useForm } from 'react-hook-form';
import { useSearchParams } from 'react-router-dom';
import { Albums, Artists, Error, Loader, Playlists } from '../../components';
import { FormData, useSearch } from '../../hooks/search';

function SearchPage(): JSX.Element {
  const [searchParams, setSearchParams] = useSearchParams();
  const defaultValues = Object.fromEntries(searchParams);
  const { register, handleSubmit } = useForm<FormData>({
    defaultValues,
  });
  const { mutate, data, error, isError, isPending } = useSearch();

  function onSubmit(formData: FormData) {
    const { q } = formData;
    setSearchParams({
      q,
    });
    mutate(formData);
  }

  if (isPending) {
    return <Loader />;
  }

  if (isError) {
    return <Error>{error.message}</Error>;
  }

  return (
    <>
      <Helmet>
        <title>Search</title>
      </Helmet>
      <div>
        <h1>Search</h1>
        <form onSubmit={handleSubmit(onSubmit)}>
          <div>
            <label htmlFor="q">Search</label>
            <input id="q" type="search" {...register('q')} />
          </div>
        </form>
        {data && (
          <>
            <h3>Tracks</h3>
            {data.tracks && (
              <ul>
                {data.tracks.items.map((track) => (
                  <li key={track.id}>{track.name}</li>
                ))}
              </ul>
            )}

            {data.artists && (
              <>
                <h3>Artists</h3>
                <Artists artists={data.artists.items} />
              </>
            )}

            {data.albums && (
              <>
                <h3>Albums</h3>
                <Albums items={data.albums.items} />
              </>
            )}

            {data.playlists && (
              <>
                <h3>Playlists</h3>
                <Playlists items={data.playlists.items} />
              </>
            )}
          </>
        )}
      </div>
    </>
  );
}

export default SearchPage;
