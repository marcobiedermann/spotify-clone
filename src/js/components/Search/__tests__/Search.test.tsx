import { render } from '@testing-library/react';
import React from 'react';
import { expect, it } from 'vitest';
import Search from '..';
import searchFixtures from '../__fixtures__';

it('renders correctly', () => {
  const { container } = render(<Search {...searchFixtures} />);

  expect(container.firstChild).toMatchSnapshot();
});
