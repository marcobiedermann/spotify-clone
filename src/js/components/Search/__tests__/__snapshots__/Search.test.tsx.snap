// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders correctly 1`] = `
<form
  class="search"
>
  <input
    class="search__input"
    type="search"
  />
</form>
`;
